public/app/components/AppSelector/AppSelector.tsx

```diff
@@ -134,6 +147,54 @@ export const SelectorModalWithToggler = ({
     return selectedApp?.__profile_type__ === a.__profile_type__;
   };
 
+  const groups = useMemo(() => {
+    const allGroups = leftSideApps.map((app) => app.name.split('-')[0]);
```

Comment:
I guess here we should maybe use `service_name` and simply calculate the height of the dropdown based off the number of `service_name`.

If I'm understanding correctly the idea here is to:
1. Calculate the necessary height of the modal based off the number of "groups" that will exist
2. If the modal is bigger than the screen then set the modal to slightly bigger height than the screen... if the modal is smaller than the screen then just adapt to the size of the modal

right now the logic is assuming that the "groups" are differentiated by the text before the first `-` but thats not true. I guess they are now based off of the `service_name`? 

cc @korniltsev  @cyriltovena  @petethepig

Reply:
this isn't super helpful because I just cheated so I could make the dropdown very high by adding random numbers to each `name`, but for production data, for example, will the height of the modal will be determined by the unique number of `service_names`